src/flake8_numbers/check_numbers.py

```diff
@@ -0,0 +1,161 @@
+
+        Args:
+            node: The AST node to extract the code from.
+
+        Returns:
+            The code of the given AST node.
+        """
+        start_line, start_col = node.lineno, node.col_offset
+        end_line, end_col = node.end_lineno, node.end_col_offset
+        with open(self._filename, "r", encoding="utf-8") as source_file:
+            lines = source_file.readlines()
+            original_code = "".join(
+                line[start_col:end_col].strip()
+                for line in lines[start_line - 1 : end_line]
+            )
+            return original_code
+
+    def _check_underscore_modulos(
+        self,
+        fragment: str,
+        original_literal: str,
+        modulo: int,
+        node: ast.Constant,
+    ) -> Optional[ErrorReport]:
+        """Check the given fragment for underscores at every modulo position.
+
+        Every part of the fragemnt that is separated by an underscore must be of length modulo.
+        The first part of the fragment is allowed to be shorter than modulo.
+
+        Args:
+            fragment: The fragment to check.
+            original_literal: The original literal that the fragment was extracted from.
+            modulo: The modulo to check for (e.g. 3 or 4).
+            node: The AST node to check.
+
+        Returns:
+            An ErrorReport if the fragment is not well formatted.
+        """
+        parts = fragment.split("_")
+        for i, part in enumerate(parts):
+            invalid_first_part = i == 0 and len(part) > modulo
+            invalid_continuation_part = i != 0 and len(part) != modulo
+            if invalid_first_part or invalid_continuation_part:
+                message = (
+                    f"NUM01: Use underscores every {modulo} digits in large numeric literals"
+                    + f" ({original_literal}) for better readability."
+                )
+                return ErrorReport(
+                    line=node.lineno,
+                    column=node.col_offset,
+                    message=message,
+                )
+
+        return None
+
+    def check_constant(self, node: ast.Constant) -> Optional[ErrorReport]:
+        """Check for the readability of the given numeric literal.
+
+        Args:
+            node: The AST node to check.
+
+        Returns:
+            An ErrorReport if the node is a number literal that is not well formatted.
+        """
+        if not isinstance(node.value, (int, float)):
+            return None
+
+        original_literal = self._extract_code(node)
+
+        if original_literal in ["True", "False"]:
+            return None
+
+        is_binary = original_literal.startswith("0b")
+        is_octal = original_literal.lower().startswith("0o")
+        is_hexadecimal = original_literal.lower().startswith("0x")
+        is_decimal = not is_binary and not is_octal and not is_hexadecimal
+        separator_modulo = 4 if is_hexadecimal else 3
```

Comment:
I privately work a lot with hexadecimal and binary numbers. I found 4 and 4 most readable. But hey, we're going to make that configurable, right? (future PR) :-)

Hmm, I think I might see why you want 2 for hex numbers, because of 0xRRGGBB values? 🤔

Reply:
As I said, because of **byte**, also if for some reason you represent ASCII (`0x30_31_32 == "012"`), it is common to reason about bytes.

But I agree there is no single answer to this like with decimals, so I guess configurability and some reasonable default is the best answer.

Reply:
Resolving as we have https://github.com/frequenz-floss/flake8-numbers/issues/6 now.